talker_tests/uts/test_client.py

```diff
@@ -0,0 +1,118 @@
+        cmd = self.client.run(self.host_id, 'bash', '-ce', 'echo hello')
+        sleep(0.1)  # write to redis must be completed before we pop from redis
+        result = self.get_command_from_redis()
+        expected_value = {'id': cmd.job_id,
+                          'cmd': ['bash', '-ce', 'echo hello'],
+                          'timeout': 3600.0}
+        self.assertEqual(result, expected_value)
+
+    def test_command_result_from_agent(self) -> None:
+        cmd = self.client.run(self.host_id, 'bash', '-ce', 'echo hello')
+        self.write_response_to_redis(cmd.job_id, 'hello\n')
+        res = cmd.result()
+        self.assertEqual(res, 'hello\n')
+
+    def test_client_wait_command(self) -> None:
+        delay = 0.5
+        cmd = self.client.run(self.host_id, 'bash', '-ce', 'sleep {}'.format(delay))
+        before = time()
+        Thread(target=self.write_response_to_redis, kwargs={'job_id': cmd.job_id, 'delay': delay}).start()
+        ret = cmd.wait()
+        after = time()
+        self.assertEqual(ret, 0)
+        self.assertTrue(delay - delay * 0.1 <= after - before <= delay + delay * 0.1)
+
+    def test_timeout_error(self) -> None:
+        delay = 5
+        cmd = self.client.run(self.host_id, 'bash', '-ce', 'sleep {}'.format(delay), timeout=delay * 0.0001,
+                              server_timeout=False)
+        Thread(target=self.write_response_to_redis, kwargs={'job_id': cmd.job_id, 'delay': delay},
+               daemon=True).start()
+        try:
+            _ = cmd.wait()
+            raise Exception('ClientCommandTimeoutError was not raised')
+        except ClientCommandTimeoutError:
+            pass
+
+    def test_command_fail(self) -> None:
+        cmd = self.client.run(self.host_id, 'bash', '-ce', 'exit 2')
+        self.write_response_to_redis(cmd.job_id, retcode=2)
+        try:
+            cmd.result()
+            raise Exception('CommandExecutionError was not raised')
+        except CommandExecutionError:
+            pass
+
+    def test_ignoring_command_fail(self) -> None:
+        cmd = self.client.run(self.host_id, 'bash', '-ce', 'exit 2', raise_on_failure=False)
+        self.write_response_to_redis(cmd.job_id, retcode=2)
+        cmd.result()
+
+    def test_logging_output(self) -> None:
+        cmd = self.client.run(self.host_id, 'bash', '-ce', 'for i in {1..5}; do date; sleep 0.05; done')
+        curr = datetime.now()
+        aggregated_logging = ''
+        for i in range(5):
+            curr += timedelta(seconds=0.1)
+            stdout_val = curr.strftime("%a %b %d %H:%M:%S IDT %Y\n")
+            self.write_response_to_redis(cmd.job_id, stdout_val=stdout_val)
+            aggregated_logging += stdout_val
+
+        log_stream = io.StringIO()
+        logging.basicConfig(stream=log_stream)
+        cmd.log_pipe(logger=logging.getLogger(), stdout_lvl=logging.WARNING)
+        output = log_stream.getvalue().replace('WARNING:root:', '')
```

Comment:
by calling `logging.getLogger()` without passing a name you will get the root logger. it is unlikely to be initialized to anything but the defaults in this session but it might. instead you can pass a name and control it's log format to fit your goals.